lib/RT/Extension/REST2/Resource/Record/Writable.pm

```diff
@@ -278,6 +278,10 @@ sub create_record {
     if ($record->isa('RT::Ticket') || $record->isa('RT::Asset')) {
         if ($cfs) {
             while (my ($id, $value) = each(%$cfs)) {
+                if ($id !~ /^\d+$/) {
+                    my $cf = $record->LoadCustomFieldByIdentifier($id);
+                    $id = $cf->Id;
+                }
                 delete $cfs->{$id};
```

Comment:
Actually the `if-condition` you've added should lie after this `delete` line, since `$id` is modified in the `if-condition`, `$cfs->{$id}` is undef and initital `$cfs->{<Name of CF>}` is not deleted. This triggers no warnings, since `RT::Ticket::Create()` will silently ignore it (doing a  `next unless $arg =~ /^CustomField-(\d+)$/i;` when parsing CFs arguments). Nevertheless, if `RT::Ticket::Create()` is changed one of these days to trigger a warning when parsing malformed CFs arguments, tests based on your pull request will fail.

Reply:
Good catch. I'll update the PR during the week.

Thanks!

Reply:
Hey @gibus , I've changed the logic, so now it applies to all objects, not just Tickets & Assets.